pkg/api/team_test.go

```diff
@@ -74,4 +77,53 @@ func TestTeamApiEndpoint(t *testing.T) {
 			})
 		})
 	})
+
+	t.Run("When creating team with api key", func(t *testing.T) {
+		defer bus.ClearBusHandlers()
+
+		sc := setupScenarioContext("/api/teams")
+		hs := &HTTPServer{
+			Cfg: setting.NewCfg(),
+			Bus: bus.GetBus(),
+		}
+		hs.Cfg.EditorsCanAdmin = true
+
+		teamName := "team foo"
+		sc.defaultHandler = Wrap(func(w http.ResponseWriter, c *models.ReqContext) Response {
```

Comment:
Do we need this handler? It seems to me if we do not want to test all the middleware stuff that we could just call `hs.CreateTeam(c, cmd)` directly without doing actual Post request later and the `setupScenarioContext`.